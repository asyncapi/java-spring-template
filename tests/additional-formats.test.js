const path = require('path');
const Generator = require('@asyncapi/generator');
const { readFile } = require('fs').promises;

const MAIN_TEST_RESULT_PATH = path.join('tests', 'temp', 'integrationTestResult');

const generateFolderName = () => {
    // you always want to generate to new directory to make sure test runs in clear environment
    return path.resolve(MAIN_TEST_RESULT_PATH, Date.now().toString());
};

describe('template integration tests for additional formats of data types', () => {

    jest.setTimeout(30000);

    it('should generate DTO file with proper type classes', async() => {
        const outputDir = generateFolderName();
        const params = {};
        const kafkaExamplePath = './mocks/additional-type-formats.yml';

        const generator = new Generator(path.normalize('./'), outputDir, { forceWrite: true, templateParams: params });
        await generator.generateFromFile(path.resolve('tests', kafkaExamplePath));

        const expectedFiles = [
            '/src/main/java/com/asyncapi/model/SongPayload.java'
        ];
        for (const index in expectedFiles) {
            const file = await readFile(path.join(outputDir, expectedFiles[index]), 'utf8');
            const fileWithAnyDate = file.replace(/date="\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d\.\d+([+-][0-2]\d:[0-5]\d|Z)"/, 'date="AnyDate"');
            expect(fileWithAnyDate).toMatchSnapshot();
        }
    });
});
