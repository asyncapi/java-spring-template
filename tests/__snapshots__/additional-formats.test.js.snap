// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template integration tests for additional formats of data types should generate DTO file with proper type classes 1`] = `
"package com.asyncapi.model;


import jakarta.validation.constraints.*;
import jakarta.validation.Valid;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonValue;

import javax.annotation.processing.Generated;
import java.util.List;
import java.util.Objects;


@Generated(value="com.asyncapi.generator.template.spring", date="AnyDate")
public class SongPayload {
    
    private @Valid java.util.UUID id;
    
    private @Valid String title;
    
    private @Valid String uri;
    
    private @Valid String email;
    
    private @Valid java.math.BigDecimal rating;
    

    

    /**
     * ID
     */
    @JsonProperty("id")
    public java.util.UUID getId() {
        return id;
    }

    public void setId(java.util.UUID id) {
        this.id = id;
    }
    

    /**
     * Title of a song
     */
    @JsonProperty("title")
    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }
    

    /**
     * URI of a song
     */
    @JsonProperty("uri")
    public String getUri() {
        return uri;
    }

    public void setUri(String uri) {
        this.uri = uri;
    }
    

    /**
     * Author email
     */
    @JsonProperty("email")
    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }
    

    /**
     * Title rating
     */
    @JsonProperty("rating")
    public java.math.BigDecimal getRating() {
        return rating;
    }

    public void setRating(java.math.BigDecimal rating) {
        this.rating = rating;
    }
    
    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        SongPayload songPayload = (SongPayload) o;
        return 
            Objects.equals(this.id, songPayload.id) &&
            Objects.equals(this.title, songPayload.title) &&
            Objects.equals(this.uri, songPayload.uri) &&
            Objects.equals(this.email, songPayload.email) &&
            Objects.equals(this.rating, songPayload.rating);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, title, uri, email, rating);
    }

    @Override
    public String toString() {
        return "class SongPayload {\\n" +
        
                "    id: " + toIndentedString(id) + "\\n" +
                "    title: " + toIndentedString(title) + "\\n" +
                "    uri: " + toIndentedString(uri) + "\\n" +
                "    email: " + toIndentedString(email) + "\\n" +
                "    rating: " + toIndentedString(rating) + "\\n" +
                "}";
    }

    /**
     * Convert the given object to string with each line indented by 4 spaces (except the first line).
     */
    private String toIndentedString(Object o) {
        if (o == null) {
           return "null";
        }
        return o.toString().replace("\\n", "\\n    ");
    }
}"
`;
